hugo/content/docs/langium-overview.md

```diff
@@ -0,0 +1,87 @@
+
+To define a custom language, users of Langium interact with a high level presentation of their context-free grammar using the [Langium grammar language](../grammar-language), in a similar fashion to EBNF. In addition to defining the syntax of your language, it allows you to embed semantic information about your model.
+
+## Semantic Models
+
+As a consequence of defining your grammar in this way, we can not only parse your language, but also generate a semantic model as TypeScript interfaces, which automatically match the parsed abstract syntax tree. The following language snippet:
+
+```ts
+Person: 
+    'person' // keyword 
+    name=ID // semantic assignment
+; 
+```
+
+Parses a simple object `{ name: 'John' }`. To interact with this object in a type safe manner, the `langium-cli` tool generates TypeScript type definitions from it. The `Person` parser rules generates the following interface:
+
+```ts
+interface Person {
+    name: String
+}
+```
+
+These interfaces allow you to traverse your abstract syntax tree safely. In case your grammar changes, they will also notify you of any breaking changes which might require you to change your domain logic.
+
+## Cross References and Linking
+
+You can reference another element of your DSL using the grammar cross reference feature:
+
+```ts
+Hello: 'hello' person=[Person];
+```
+
+This `Hello` rule references another `Person` element. By default, the `name` property of an object is used as its identity. Consequently, a `person John` object will be referenced using the `hello John` snippet. Any other name will result in a linking error.
+
+Cross references and linking is accomplished using the concept of 'scoping'. You likely know scopes from programming, where certain variables are only available from certain scopes:
+
+```ts
+let x = 42;
+x = 3; // References the `x` defined in the previous line
+```
+
+```ts
+if (something) {
+    let y = 42;
+}
+y = 3; // Cannot link, `y` isn't in a visible scope
+```
+
+The same occurs in Langium. To enable more complex scoping behavior, you can create a custom domain scoping. For example, common object oriented.languages need a more involved scoping mechanism to resolve references to fields and methods of a class:
+
+```ts
+class X {
+    y(): void { ... }
+}
+
+const instance = new X(); // Symbol `X` is in the local scope
+instance.y(); // Symbol `y` exists in the scope of the `X` class
+```
+
+Aside from those domain specific scoping rules, parsing and resolving cross references is handled completely automatically by Langium.
+
+## Workspace Management
+
+Like with common modularized programming languages, domain logic written in your DSL will usually be split across multiple files to facilitate ease of use and maintenance efforts. This is also possible using Langium, which performs workspace management for you.
```

Comment:
In the same idea as the other sections, I would really be good to start clear and begin with sentence describing how Langium helps with workspace management. Following that, it would be also good to describe/define your usage of `workspace` in this context. That will help readers discern whether they're following you correctly, or if they're thinking of something else.